File: advent2024/a01.py
import sys
import collections

with open(sys.argv[1]) as f:
    data = f.read()

nums1 = []
nums2 = []
for line in data.split('\n'):
    if not line:
        continue
    n1, n2 = line.split()
    n1 = int(n1)
    n2 = int(n2)
    nums1.append(n1)
    nums2.append(n2)

snums1 = sorted(nums1)
snums2 = sorted(nums2)
d = 0

for n1, n2 in zip(snums1, snums2):
    d += abs(n1 - n2)

print(d)

s = 0
counts = collections.Counter(nums2)
for n1 in nums1:
    s += n1 * counts[n1]

print(s)

File: advent2024/a01.pl
use 5.10.0;

my $f;
my @nums1;
my @nums2;
my %counts;

open $f, '<', @ARGV[0] || die 'file pls';

while(<$f>) {
    my @vals = split /\s+/, $_;
    push @nums1, @vals[0] + 0;
    push @nums2, @vals[1] + 0;
    $counts{@vals[1] + 0} += 1;
}

close $f;

my @snums1 = sort @nums1;
my @snums2 = sort @nums2;
my $d = 0;
my $s = 0;

for (my $i = 0; $i < @nums1; $i++) {
    $d += abs(@snums1[$i] - @snums2[$i]);
    $s += @nums1[$i] * $counts{@nums1[$i]};
}

say $d;
say $s;
